stablehlo/tests/interpret_iota.mlir

```diff
@@ -0,0 +1,1213 @@
+// RUN: stablehlo-interpreter --interpret -split-input-file %s | FileCheck %s
+
+// CHECK-LABEL: Evaluated results of function: iota_op_test_si4_dim_0
+func.func @iota_op_test_si4_dim_0() -> tensor<3x4xi4> {
+  %0 = stablehlo.iota dim = 0 : tensor<3x4xi4>
+  func.return %0 : tensor<3x4xi4>
+  // CHECK-NEXT: tensor<3x4xi4>
+  // CHECK-NEXT: 0 : i4
+  // CHECK-NEXT: 0 : i4
+  // CHECK-NEXT: 0 : i4
+  // CHECK-NEXT: 0 : i4
+  // CHECK-NEXT: 1 : i4
+  // CHECK-NEXT: 1 : i4
+  // CHECK-NEXT: 1 : i4
+  // CHECK-NEXT: 1 : i4
+  // CHECK-NEXT: 2 : i4
+  // CHECK-NEXT: 2 : i4
+  // CHECK-NEXT: 2 : i4
+  // CHECK-NEXT: 2 : i4
+}
+
+// CHECK-LABEL: Evaluated results of function: iota_op_test_si4_dim_1
+func.func @iota_op_test_si4_dim_1() -> tensor<3x4xi4> {
+  %0 = stablehlo.iota dim = 1 : tensor<3x4xi4>
+  func.return %0 : tensor<3x4xi4>
+  // CHECK-NEXT: tensor<3x4xi4>
+  // CHECK-NEXT: 0 : i4
+  // CHECK-NEXT: 1 : i4
+  // CHECK-NEXT: 2 : i4
+  // CHECK-NEXT: 3 : i4
+  // CHECK-NEXT: 0 : i4
+  // CHECK-NEXT: 1 : i4
+  // CHECK-NEXT: 2 : i4
+  // CHECK-NEXT: 3 : i4
+  // CHECK-NEXT: 0 : i4
+  // CHECK-NEXT: 1 : i4
+  // CHECK-NEXT: 2 : i4
+  // CHECK-NEXT: 3 : i4
+}
+
+// CHECK-LABEL: Evaluated results of function: iota_op_test_si4_dim_0_overflow
+func.func @iota_op_test_si4_dim_0_overflow() -> tensor<16x1xi4> {
+  %0 = stablehlo.iota dim = 0 : tensor<16x1xi4>
+  func.return %0 : tensor<16x1xi4>
+  // CHECK-NEXT: tensor<16x1xi4>
+  // CHECK-NEXT: 0 : i4
+  // CHECK-NEXT: 1 : i4
+  // CHECK-NEXT: 2 : i4
+  // CHECK-NEXT: 3 : i4
+  // CHECK-NEXT: 4 : i4
+  // CHECK-NEXT: 5 : i4
+  // CHECK-NEXT: 6 : i4
+  // CHECK-NEXT: 7 : i4
+  // CHECK-NEXT: -8 : i4
+  // CHECK-NEXT: -7 : i4
+  // CHECK-NEXT: -6 : i4
+  // CHECK-NEXT: -5 : i4
+  // CHECK-NEXT: -4 : i4
+  // CHECK-NEXT: -3 : i4
+  // CHECK-NEXT: -2 : i4
+  // CHECK-NEXT: -1 : i4
+}
+
+// CHECK-LABEL: Evaluated results of function: iota_op_test_si4_dim_1_overflow
+func.func @iota_op_test_si4_dim_1_overflow() -> tensor<1x16xi4> {
+  %0 = stablehlo.iota dim = 1 : tensor<1x16xi4>
+  func.return %0 : tensor<1x16xi4>
+  // CHECK-NEXT: tensor<1x16xi4>
+  // CHECK-NEXT: 0 : i4
+  // CHECK-NEXT: 1 : i4
+  // CHECK-NEXT: 2 : i4
+  // CHECK-NEXT: 3 : i4
+  // CHECK-NEXT: 4 : i4
+  // CHECK-NEXT: 5 : i4
+  // CHECK-NEXT: 6 : i4
+  // CHECK-NEXT: 7 : i4
+  // CHECK-NEXT: -8 : i4
+  // CHECK-NEXT: -7 : i4
+  // CHECK-NEXT: -6 : i4
+  // CHECK-NEXT: -5 : i4
+  // CHECK-NEXT: -4 : i4
+  // CHECK-NEXT: -3 : i4
+  // CHECK-NEXT: -2 : i4
+  // CHECK-NEXT: -1 : i4
+}
+
```

Comment:
You may add the split markers. Just to follow what other tests are doing. Not required though.